libraries/AP_Notify/ToneAlarm.cpp

```diff
@@ -110,6 +112,18 @@ bool AP_ToneAlarm::init()
     flags.pre_arm_check = 1;
     _cont_tone_playing = -1;
     hal.scheduler->register_timer_process(FUNCTOR_BIND(this, &AP_ToneAlarm::_timer_task, void));
+
+#if (HAL_OS_POSIX_IO || HAL_OS_FATFS_IO) && defined(HAL_BOARD_STORAGE_DIRECTORY)
```

Comment:
Does this work appropriately on linux/sitl?  It looks like, depending on initialisation order, you might get this tone out on BeBop (https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_HAL/board/linux.h#L38), for example, even 'though Storage magically creates things later (https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_HAL_Linux/Storage.cpp#L151).

Reply:
I think I'll make it ChibiOS only, as it doesn't really make sense to play "no storage" on Linux